Comments / formatting

JBBianchi opened the pull request:
**What this PR does / why we need it**:
Improves code quality and consistancy with linting and prettifying.
Adds comments & file headers.
Fixes builder problems.

Linked to #79 
Linked to #94 

Closes #89
Closes #90
Closes #91
Closes #92
Closes #93

JBBianchi commented:
Not sure husky/lint-staged work as expected...

JBBianchi commented:
@antmendoza @tsurdilo are you ok with merging this ? I'm not a 100% sure about the husky/lint-staged config but it's a step forward anyways and *shouldn't* break anything :)

antmendoza commented:
I will review it today, thank you @JBBianchi

antmendoza commented:
Hi @JBBianchi

I think that husky is not working for me if I am getting it right. 

The pre-commit hook will run "npx lint-staged", which run "npm run format" for all the staged files, isn't it?

In my case nothing happen

any suggestions ?

JBBianchi commented:
> Hi @JBBianchi
> 
> I think that husky is not working for me if I am getting it right.
> 
> The pre-commit hook will run "npx lint-staged", which run "npm run format" for all the staged files, isn't it?
> 
> In my case nothing happen
> 
> any suggestions ?

lint-staged will run "npm run format" if there is any .ts staged if I understood well but I must admit I don't think it's working.

JBBianchi commented:
I fixed lint-staged/husky I think. 1st, I forgot to install husky properly, 2nd I didn't use the proper command to format (`npm run format` is formatting all the files, with lint-staged we want to only format the staged ones. So in the end, I just used `prettier --write`).

I also added file header and some fixes:
#89
#90
#91
#92
#93
#94

antmendoza commented on README.md:
```diff
@@ -1,152 +1,197 @@
+import { validators, Specification } from '@severlessworkflow/sdk-typescript';
+
+const injectionState: Specification.Injectstate = workflow.states[0];
+const injectionStateValidator: ValidateFunction<Specification.Injectstate> = validators.get('Injectstate');
```
`+` if I where a user of the sdk, I don't want to learn the keys for each element to retrieve the validator from `validators`

JBBianchi replied:
The 'keys' are the type names of Specification but I agree with you, it could be improved. I need to double check but I don't think we can use the type directly in TS like `validator.get(Specification.Injectstate)` because the type doens't exist at runtime. Anyhow, good idea, there is room for improvement here.

One side note though, more regarding the schema, I think Events/Retries/Functions cannot be used by themselves because they are based on custom properties that don't exist in the JSON Schema specification. So they can be "pointed at" but not used alone. I might be wrong though. (See little chat in the [JSON Schema Slack](https://app.slack.com/client/T5CHC2TAN/C5CF75URH/thread/C5CF75URH-1619796577.402700))

antmendoza commented on tools/consts.ts:
```diff
@@ -0,0 +1,21 @@
+export const fileHeader = `/*
+* Copyright 2021-Present The Serverless Workflow Specification Authors
```
👍

antmendoza commented on .husky/.gitignore:
```diff
@@ -0,0 +1 @@
+_
```
I am just curious, what this is supposing to do?

JBBianchi replied:
It's added by husky. When doing `husky install`, it creates `.husky/` with that `.gitignore` and a `_/` with a .sh in it.

antmendoza reviewed (commented):
husky is not working for me, no idea why. 

Instead it works adding the pre-commit hook like it shows the documentation here https://www.npmjs.com/package/husky 
`npx husky add .husky/pre-commit "npm test"` 

Could you check if this is working also for you?

I am gonna merge this anyway, we can fix it in another pr. 


Thank you @JBBianchi

JBBianchi commented:
> husky is not working for me, no idea why.
> 
> Instead it works adding the pre-commit hook like it shows the documentation here https://www.npmjs.com/package/husky
> `npx husky add .husky/pre-commit "npm test"`
> 
> Could you check if this is working also for you?
> 
> I am gonna merge this anyway, we can fix it in another pr.
> 
> Thank you @JBBianchi

Maybe you need to run the `husky install` ? It creates `.husky/_/` containing a `husky.sh`. I'll double check tomorrow, I'm not familiar with husky/lint-staged.

JBBianchi commented on .github/ISSUE_TEMPLATE/bug-report.md:
```diff
@@ -1,16 +1,16 @@
-
-**Anything else we need to know?**:
-
+---
```
prettier supports `MD` indeed but I don't really understand why they have been formatted... I thought I specified `.ts` only...

tsurdilo commented:
is there anything I can help with this? if you have some steps i can follow can try on a clean box

JBBianchi commented:
@tsurdilo I would be nice if you could get the project, run `npm install`, create a new dummy .ts file with something like `export const foo = "bar";` and commit locally. 

Huksy should kick in and, in the end, the file should be prettyfied as `export const foo = 'bar';`  (with single quotes)

JBBianchi commented:
I think @antmendoza is right, they changed the way to handle hooks. The js/json/package method has been replaced by the .husky directory thing (https://blog.typicode.com/husky-git-hooks-javascript-config/). I'll fix that.
